### ISSUE TITLE
Youtube DL not downloading to desktop

### ISSUE THREAD

**Issue by ghost** [ytdl-org/youtube-dl#31646]
Unable to extract uploader id; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update.

I just want to download a music video to my desktop. But the commands are unable to extract and also like the other codes are being rejected

**Comment by hexisXz** [comment]
i just got the same this try this 

sudo wget https://yt-dl.org/downloads/latest/youtube-dl -O /usr/local/bin/youtube-dl
sudo chmod a+rx /usr/local/bin/youtube-dl
hash -r
then run sudo youtube-dl -U and if gives a python error just add a 3 after python in sudo vim /usr/local/bin/youtube-dl
so its python3 then run sudo youtube-dl -U again and try to download it again then it should work but thats all i got rn

**Comment by ananay** [comment]
@hexisXz I tried that but that doesn't work. The only way I could get it to work was cloning the repository, `cd`ing into the cloned repository and running using `python3 -m youtube_dl -x --audio-format mp3 <URL>`.

**Comment by comicsansgreenki** [comment]
@lorettaimmaculatem Please see #31530, and #30839 if you haven't yet. Thanks!

(sidenote: @ananay is the `--audio-format mp3` actually needed for your extraction to work? afaik it just gets ffmpeg to turn whatever into an mpeg3 file.)

**Comment by ananay** [comment]
> @lorettaimmaculatem Please see #31530, and #30839 if you haven't yet. Thanks!
> 
> (sidenote: @ananay is the `--audio-format mp3` actually needed for your extraction to work? afaik it just gets ffmpeg to turn whatever into an mpeg3 file.)

Nope, just an example to convert to audio. It isn't really required to get around this specific issue. I can edit my comment for that.

**Comment by comicsansgreenki** [comment]
> Nope, just an example to convert to audio. It isn't really required to get around this specific issue. I can edit my comment for that.

ah yeah, ok. was making sure, cuz i'd hate to lead other users the wrong way and create issues complaining why their downloads kept popping up as audio-only.

**Comment by ananay** [comment]
> > Nope, just an example to convert to audio. It isn't really required to get around this specific issue. I can edit my comment for that.
> 
> ah yeah, ok. was making sure, cuz i'd hate to lead other users the wrong way and have them create issues asking why their downloads kept popping up as audio-only.

Makes sense, thanks for pointing that out!

**Comment by lorettaimmaculatem** [comment]
Thank you all! It worked out well!

**Comment by hexisXz** [comment]
> @hexisXz I tried that but that doesn't work. The only way I could get it to work was cloning the repository, `cd`ing into the cloned repository and running using `python3 -m youtube_dl -x <URL>`.

how do i do it with video?

**Comment by hexisXz** [comment]
i mean @ananay

**Comment by ananay** [comment]
@hexisXz you can use `python3 -m youtube_dl <URL>` [Remove the `-x` basically] and you can get the video itself. Let me know if you have any questions

**Comment by comicsansgreenki** [comment]
@hexisXz `youtube_dl <URL>` will pull the best video quality available on its own and transcode it to mp4 (if you have ffmpeg installed). 
if you're looking for a specific format, try `youtube_dl -F <URL>` - this will list all available formats on the media you're requesting. when you pick your format, add the option `-f <format_code>` to the command.

**Comment by Vangelis66** [comment]
> `youtube_dl <URL>` will pull the **best video quality available on its own** and **transcode** it to mp4 (if you have ffmpeg installed, and if not already mp4).

Actually, the **(best)** format, downloaded by that simple sample command, is determined by the availability of a **standalone media container** (usually MP4) with BOTH video+audio elementary streams, delivered over HTTPS-progressive... For `youtube.com` streams, this may result currently in `360p` files, when higher resolutions are available/desired...

If a standalone container with both AV content is unavailable, then **(best)** may be a fragmented video+audio stream, delivered over HLS or DASH (rarely, over MS's "SmoothStreaming"); `ffmpeg` is required to "**remux**" the downloaded stream to a container palatable to most media players, usually MP4 (or MKV or WebM, a `matroska` subspecies) ... 
Likewise, if `video-only` or `audio-only` streams are served by the service of choice, `ffmpeg` is used to "combine" ("**mux**" is the proper term) the independently downloaded **raw** video and audio streams into a **container**... 
In all the above cases, **TRANSCODING**/RECODING isn't performed **by default**, which is quite a CPU-intensive and time-consuming process in itself... 

Transcoding/re-encoding (by `ffmpeg`) may be configured to take place after download, via a user-set option (e.g `--recode-video` or some `--postprocessor-args` command...). 

**TL;DR**: [re]muxing != transcoding

Best regards.
